# RAW DATASET EXPLORATION

In [ ]:
import h5py

file_path = "../data/raw/background_for_training.h5"

with h5py.File(file_path, "r") as f:
    print(list(f.keys()))
    print('\n')

    particles = f["Particles"]
    print(particles)
    print(particles[:10,:,:])
    print('\n')

    classes = f["Particles_Classes"]
    print(classes)
    print(classes[:])
    print('\n')

    names = f["Particles_Names"]
    print(names)
    print(names[:])

['Particles', 'Particles_Classes', 'Particles_Names']


<HDF5 dataset "Particles": shape (13451915, 19, 4), type "<f8">
[[[ 2.74129639e+01  0.00000000e+00 -2.04990220e+00  1.00000000e+00]
  [ 2.42099686e+01  1.63350999e+00 -1.03496209e-01  2.00000000e+00]
  [ 0.00000000e+00  0.00000000e+00  0.00000000e+00  0.00000000e+00]
  [ 0.00000000e+00  0.00000000e+00  0.00000000e+00  0.00000000e+00]
  [ 0.00000000e+00  0.00000000e+00  0.00000000e+00  0.00000000e+00]
  [ 0.00000000e+00  0.00000000e+00  0.00000000e+00  0.00000000e+00]
  [ 0.00000000e+00  0.00000000e+00  0.00000000e+00  0.00000000e+00]
  [ 0.00000000e+00  0.00000000e+00  0.00000000e+00  0.00000000e+00]
  [ 0.00000000e+00  0.00000000e+00  0.00000000e+00  0.00000000e+00]
  [ 0.00000000e+00  0.00000000e+00  0.00000000e+00  0.00000000e+00]
  [ 0.00000000e+00  0.00000000e+00  0.00000000e+00  0.00000000e+00]
  [ 0.00000000e+00  0.00000000e+00  0.00000000e+00  0.00000000e+00]
  [ 0.00000000e+00  0.00000000e+00  0.00000000e+00  0.00000000e+

## HDF5 File Structure

| Name | Shape | Type |
|---|---|---|
| Particles | (13451915, 19, 4) | float |
| Particles_Classes | (4,) | string |
| Particles_Names | (4,) | string |

### Particles

Shape: **(N_events, N_particles, N_features) = (13451915, 19, 4)**

Each event (13451915 total) is a 19 × 4 matrix:
- 19 is the number of particles per event
- 4 are the features per particle (Particles_Names)

Slots not occupied by a real particle are filled with zeros across all 4 features, to have fixed-size inputs for all the different events.

### Particles_Names

| Index | Name | Description |
|---|---|---|
| 0 | Pt | Transverse momentum (GeV) |
| 1 | Eta | Pseudorapidity η |
| 2 | Phi | Azimuthal angle φ (rad) |
| 3 | Class | Particle type (Particles_Classes) |

### Particles_Classes

| Class value | Name | Description |
|---|---|---|
| 1.0 | MET_class_1 | Missing Transverse Energy |
| 2.0 | Four_Ele_class_2 | Electrons |
| 3.0 | Four_Mu_class_3 | Muons |
| 4.0 | Ten_Jet_class_4 | Hadronic Jets |

Each event has always 19 lines. Line 0 is the MET, lines 1 to 4 are electrons ordered with decreasing Pt, 5 to 8 are muons and 9 to 18 are jets.